WeatherDataFetcher.py

```diff
@@ -1,101 +1,166 @@
+    ---------
+    location_coords : Tuple[int:int]
+      (lat,long) coordinates for the city entered
+    recent_data_available : bool
+      if true, use cached values (load from file)
+      if false, reload from url (requests)
 
+    Returns: None
+    """
+
+    response = requests.get(
+        f"https://api.openweathermap.org/data/2.5/forecast?",
+        params={
+            "lat": location_coords[0],
+            "lon": location_coords[1],
+            "dt": 25,
+            "units": "metric",
+            "appid": api_key,
+        },
+    )
+
+    # Check if fetch_data_from_url is necessary
+    file_path_json = os.getcwd() + "/data.json"
+    fetch_data_from_url = False
+    if not os.path.exists(file_path_json):
+        fetch_data_from_url = True
+    else:
+        with open(file_path_json, "r", encoding="utf-8") as f:
+            data = json.loads(f.read())
+            now_utc_in_s = datetime.datetime.now(datetime.timezone.utc).timestamp()
+            oldest_utc_in_s = data["list"][0]["dt"]
+            # Data is more than day old - fetch_data_from_url required
+            if now_utc_in_s - oldest_utc_in_s > 86400:
+                fetch_data_from_url = True
+            # Different location is requested - fetch_data_from_url required
+            lat = data["city"]["coord"]["lat"]
+            lon = data["city"]["coord"]["lon"]
+            threshold = 0.001
+            if abs(lat - location_coords[0]) > threshold:
+                fetch_data_from_url = True
+            if abs(lon - location_coords[1]) > threshold:
+                fetch_data_from_url = True
+
+    if not os.path.exists(file_path_json) or fetch_data_from_url:
+        print("Getting new data from URL", file=sys.stderr)
+        data = response.json()
+        with open(file_path_json, "w", encoding="utf-8") as f:
+            json.dump(data, f, ensure_ascii=False, indent=4)
+    else:
+        print("Getting new data local file", file=sys.stderr)
+        with open(file_path_json, "r", encoding="utf-8") as f:
+            data = json.loads(f.read())
+
+    time_to_max_temp = {}
+    time_to_min_temp = {}
+    for _, values in enumerate(data["list"]):
+        timestamp = datetime.datetime.utcfromtimestamp(values["dt"])
+        time_to_max_temp[str(timestamp)] = values["main"]["temp_max"]
+        time_to_min_temp[str(timestamp)] = values["main"]["temp_min"]
+
+    output = {}
+    print("Temperatures: ", file=sys.stderr)
+    output["max_temperatures"] = time_to_max_temp
+    output["min_temperatures"] = time_to_min_temp
+    print(output)
+
+
+def fetch_location_coords(city: str) -> Tuple[int, int]:
+    """
+    Gets the (lat,long) coordinates of a city name. If the city cannot be found
+    (e.g. invalid user input) the user is asked to input it again.
+    Parameters:
+    ----------
+    city : str
+      Name of the city to get the coordinates for
 
+    Returns:
+    ----------
+    location_coords : Tuple[int:int]
```

Comment:
The `:` wants to be a comma

```suggestion
    location_coords : Tuple[int, int]
```